pkg/persistence/postgrespersister.go

```diff
@@ -177,65 +177,41 @@ func (p *PostgresPersister) CreateIndices() error {
 	return err
 }
 
-func (p *PostgresPersister) updateDBQueryBuffer(updatedFields []string, tableName string, dbModelStruct interface{}) (bytes.Buffer, error) {
-	var queryBuf bytes.Buffer
-	queryBuf.WriteString("UPDATE ") // nolint: gosec
-	queryBuf.WriteString(tableName) // nolint: gosec
-	queryBuf.WriteString(" SET ")   // nolint: gosec
-	for idx, field := range updatedFields {
-		dbFieldName, err := cpostgres.DbFieldNameFromModelName(dbModelStruct, field)
+func (p *PostgresPersister) persisterVersionFromTable(tableName string) (*string, error) {
+	if p.version == nil {
+		version, err := p.retrieveVersionFromTable(tableName)
 		if err != nil {
-			return queryBuf, fmt.Errorf("Error getting %s from %s table DB struct tag: %v", field, tableName, err)
-		}
-		queryBuf.WriteString(fmt.Sprintf("%s=:%s", dbFieldName, dbFieldName)) // nolint: gosec
-		if idx+1 < len(updatedFields) {
-			queryBuf.WriteString(", ") // nolint: gosec
+			return nil, err
 		}
+		p.version = version
 	}
-	return queryBuf, nil
+	return p.version, nil
 }
 
-func (p *PostgresPersister) retrieveVersionFromTable(tableName string) (string, error) {
-	dbVersionData := []postgres.VersionData{}
-	queryString := fmt.Sprintf(`SELECT * FROM %s WHERE service_name=$1;`, tableName) // nolint: gosec
-	err := p.db.Select(&dbVersionData, queryString, crawlerServiceName)
+func (p *PostgresPersister) retrieveVersionFromTable(tableName string) (*string, error) {
```

Comment:
Changed this so that we save all versions to table, and only retrieve the most recent version